### ISSUE TITLE
io_uring_prep_setsockopt

### ISSUE THREAD

**Issue by sharpobject** [axboe/liburing#763]
Hello,

It would be useful to be able to call setsockopt asynchronously because TCP_QUICKACK has to be set repeatedly during the life of a connection (what the heck!)

**Comment by JonKohler** [comment]
This brings up a deeper question - I've been wondering, would it be worth while (or even possible) to have a "permanent quickack" setting in the kernel? 

the need to set quickack constantly in situations where it is configured seems so heavy handed

**Comment by axboe** [comment]
It would not be hard to add getsockopt/setsockopt support now, since we have the fops->uring_cmd() passthrough available. It's just used for nvme passthrough right now, but plumbing it up for sockets would be trivial. There are a few basic examples at the top of this branch:

https://git.kernel.dk/cgit/linux-block/log/?h=io_uring-fops.v7

I'll take a look at this when I get back from break.

**Comment by sharpobject** [comment]
> This brings up a deeper question - I've been wondering, would it be worth while (or even possible) to have a "permanent quickack" setting in the kernel?

Yes, absolutely. If we had that, it would still be useful to be able to do all our connection setup via io_uring though.

**Comment by illiliti** [comment]
Another usecase for io_uring_prep_setsockopt is IP[V6]_PKTINFO option allows setting interface and address through which messages will be sent. This option is extremely useful for high performance udp servers that listen on wildcard address(dns server is example).

**Comment by axboe** [comment]
Been available for a while.
